### PR TITLE
Gutenboarding: Update mobile header

### LINKED ISSUE #40899: Gutenboarding: mobile header
In [Gutenboarding](https://wpcalypso.wordpress.com/new), the header should look like this on mobile:

Screenshot 2020-04-08 at 14 07 55

Screenshot 2020-04-08 at 14 08 09

Screenshot 2020-04-08 at 14 08 26

### PR DESCRIPTION
**Pull request by razvanpapadopol**
#### Changes proposed in this Pull Request

* display 'Domain available' instead of recommended domain name
* change domain picker button color and font-size to match design (also for desktop)

#### Testing instructions

* go to /new on mobile, enter a site title and compare header with #40899

Fixes #40899

### CODE REVIEW

File: client/landing/gutenboarding/gutenberg-types-patch.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import '@wordpress/compose';
+import '@wordpress/element';
+
```
Comment (sirreal) [inline client/landing/gutenboarding/gutenberg-types-patch.d.ts:6]: Was it necessary to import these?

Comment (razvanpapadopol) [reply client/landing/gutenboarding/gutenberg-types-patch.d.ts:6]: Declarations seems to be ignored by TS if there is no import/export inside the module. However only one would be enough so it doesn't matter what we import.

If I do an empty export like at the end of [this example(https://www.typescriptlang.org/docs/handbook/declaration-files/templates/global-modifying-module-d-ts.html), typecheck is running fine but when building the app I get this error from webpack `Uncaught Error: Cannot find module '../../gutenberg-types-patch'`

**Comment by razvanpapadopol** [comment]
> The code LGTM. But If the intention is to make sure the sentence fits, this fix won't cover the case after the domain is picked. Maybe we should consult with @dubielzyk.

This fixed it: e4e6a58

**Comment by dubielzyk** [comment]
Yep, let's align the domain left :)

File: client/landing/gutenboarding/gutenberg-types-patch.d.ts (hunk 1 above)
Comment (sirreal) [reply client/landing/gutenboarding/gutenberg-types-patch.d.ts]: Nice find 👍

File: client/landing/gutenboarding/components/login-form/index.tsx [hunk 2]
```diff
@@ -15,14 +15,8 @@ import EnterUsernameOrEmailForm from './enter-username-or-email';
...
-// TODO: deploy this change to @types/wordpress__element
-declare module '@wordpress/element' {
-	// eslint-disable-next-line no-shadow
-	export function createInterpolateElement(
-		interpolatedString: string,
-		conversionMap: Record< string, ReactElement >
-	): ReactNode;
-}
+// TODO: remove after @types/wordpress__element update
+/*#__PURE__*/ import '../../gutenberg-types-patch';
```
Comment (sirreal) [inline client/landing/gutenboarding/components/login-form/index.tsx:19]: I believe we should do this 1 time and have the type available throughout. Does it work if we do it in `client/gutenboarding/landing/index` and not everywhere we need these types?

File: client/landing/gutenboarding/components/header/index.tsx [hunk 3]
```diff
@@ -126,10 +130,12 @@ const Header: FunctionComponent = () => {
...
-			{ recommendedDomainSuggestion
-				? /* translators: domain name is available, eg: "yourname.com is available" */
-				  sprintf( __( '%s is available' ), recommendedDomainSuggestion.domain_name )
-				: 'example.wordpress.com' }
+			{ isMobile && __( 'Domain available' ) }
+			{ ! isMobile &&
+				( recommendedDomainSuggestion
+					? /* translators: domain name is available, eg: "yourname.com is available" */
+					  sprintf( __( '%s is available' ), recommendedDomainSuggestion.domain_name )
+					: 'example.wordpress.com' ) }
```
Comment (sirreal) [inline client/landing/gutenboarding/components/header/index.tsx:138]: I'd love a beautiful nested ternary here, but the linter will disagree with my sense of aesthetics 🙃

**Comment by alshakero** [comment]
I think we need to hide the excess. See this: 

hash-e4e6a58f373a6f5bb71bfeb756d17ec6bc762e7d calypso live_new(Galaxy S5) (1)

**Comment by sirreal** [comment]
> I think we need to hide the excess. See this:
> 
> hash-e4e6a58f373a6f5bb71bfeb756d17ec6bc762e7d calypso live_new(Galaxy S5) (1)

We might need to use a larger breakpoint. You shouldn't see the domain name on the mobile header.

**Comment by simison** [comment]
@dubielzyk how do we signal that domain has been chosen in mobile header?

Unselected state is:

image

Should we cut long titles and ensure "domain available" text always fits in (it could be longer in some languages)?

File: client/landing/gutenboarding/components/header/index.tsx (hunk 3 above)
Comment (razvanpapadopol) [reply client/landing/gutenboarding/components/header/index.tsx:135]: also this format helps preserving some basic level of readability once we start adding ternaries to the mobile case 🤷

**Comment by razvanpapadopol** [comment]
> Should we cut long titles and ensure "domain available" text always fits in (it could be longer in some languages)?

We shouldn't worry about "Domain available" translation.

Screenshot 2020-04-15 at 13 31 59

I think we can limit the domain name displayed inside the button to something like 70 characters. Theoretically it can be up to 253 characters long ([255 octets](https://tools.ietf.org/html/rfc1034))

**Comment by dubielzyk** [comment]
I'll have a look

**Comment by razvanpapadopol** [comment]
Thanks everyone for reviewing this! 💯  
I'll be merging it now and opened #41133 as a follow up to track the long domain name issue.

**Pull request merged**